src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,250 @@
+
+  h.Resize(N, kCopyData);
+  srfft_->Compute(h.Data(), true);
+
+  Vector<float> temp_pad(M-1);
+  temp_pad.SetZero();
+  Vector<float> x_block_padded(N);
+
+  for (int po = 0; po < xdim; po += L) {
+    // get block of x
+    int remain = xdim - po;
+    int l = L;
+    if ( remain < l) l = remain;
+    SubVector<float> x_block(x, po, l);
+    x_block_padded.SetZero();
+    SubVector<float> ttt(x_block_padded, 0, l);
+    ttt.CopyFromVec(x_block);
+
+    srfft_->Compute(x_block_padded.Data(), true);
+
+    for (int i = 0; i < N / 2; i++) {
+      float tmp = x_block_padded(2*i);
+      x_block_padded(2*i) = tmp * h(2*i) - x_block_padded(2*i+1) * h(2*i+1);
+      x_block_padded(2*i+1) = tmp * h(2*i+1) + x_block_padded(2*i+1) * h(2*i);
+    }
+
+    srfft_->Compute(x_block_padded.Data(), false);
+    x_block_padded.Scale(1.0/N);
+
+    // combine the block
+    if (po + L < xdim) {       // current block is not the last block
+      for (int j = 0; j < L; j++)
+        x(po + j) = x_block_padded(j);
+      for (int j = 0; j < M-1; j++)
+        x(po + j) += temp_pad(j);
+      for (int j = 0; j < M-1; j++)
+        temp_pad(j) = x_block_padded(j + L);
+    } else {
+      for (int j = 0; j < xdim - po; j++)
+        x(po + j) = x_block_padded(j);
+      for (int j = 0; j < M-1 && j < xdim - po; j++)
+        x(po + j) += temp_pad(j);
+    }
+  }
+}
+
+void RepeatedlyAdd(Vector<BaseFloat> &x, Vector<BaseFloat> &h) {
+  for (int po = 0; po < x.Dim(); po += h.Dim()) {
+    int remain = x.Dim() - po;
+    int l = h.Dim();
+    if ( remain < l) l = remain;
+    SubVector<float> x_block(x, po, l);
+    SubVector<float> subh(h, 0, l);
+    x_block.AddVec(1.0, subh);
+  }
+}
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+
+    const char *usage =
+        "Corrupts the wave files supplied via input pipe with the specified\n"
+        "room-impulse response (RIR) and additive noise distortions\n"
+        "(specified by corresponding files).\n";
+
+    ParseOptions po(usage);
+    std::string noise_file;
+    BaseFloat snr_db = 20;
+
+    po.Register("noise-file", &noise_file,
+                "file with additive noise");
+    po.Register("snr-db", &snr_db,
+                "desired SNR(dB) of the output");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() != 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    std::string source_file = po.GetArg(1);
+    std::string rir_file = po.GetArg(2);
+    std::string out_file = po.GetArg(3);
+
+    WaveData src_wave;
+    {
+      Input ki(source_file);
+      src_wave.Read(ki.Stream());
+    }
+    const Matrix<BaseFloat> &SRC = src_wave.Data();
+    BaseFloat samp_freq_src = src_wave.SampFreq();  // read sampling fequency
+    int32 num_samp_src = SRC.NumCols(),  // number of samples in recording
+          num_chan_src = SRC.NumRows();  // number of channels in recording
+
+    KALDI_LOG << "samp_freq of source is " << samp_freq_src
```

Comment:
I think it is added it @danpovey's grandiose nnet3 merge couple days ago. Keep this change until  right after LGTM on all other points, since if you just pull master into your branch now the PR page will blow up to an unwieldy size.